Assets/Scripts/Screeps_API/ServerCache.cs

```diff
@@ -10,23 +10,41 @@ public class ServerCache
 
         public Address Address = new Address();
         public Credentials Credentials = new Credentials();
-        [Obsolete("We should not cache terrain on disk")] // can't remove it without breaking deserilization, have to figure out how to write some migration
+
+        [Obsolete(
+            "We should not cache terrain on disk")]
+        // can't remove it without breaking deserialization, have to figure out how to write some migration
         public Dictionary<string, string> Terrain = new Dictionary<string, string>();
+
         public bool SaveCredentials;
 
         public int LikeCount { get; set; }
+
+        [Obsolete("This boolean is not needed anymore, kept for compatibility See ServerType.")]
         public bool MMO { get; internal set; }
+
+        public ServerType Type { get; internal set; }
         public bool? Online { get; internal set; }
         public int Users { get; internal set; }
         public string Version { get; internal set; }
+
         public bool HasCredentials
         {
             get
             {
-                return !string.IsNullOrEmpty(Credentials.Token) || (!string.IsNullOrEmpty(Credentials.Email) && !string.IsNullOrEmpty(Credentials.Email));
+                return !string.IsNullOrEmpty(Credentials.Token) || !string.IsNullOrEmpty(Credentials.Email) &&
+                       !string.IsNullOrEmpty(Credentials.Email);
             }
         }
 
         public bool Selected { get; internal set; }
     }
+
+    public enum ServerType
+    {
+        NONE,
+        Official,
+        Community,
+        Custom
+    }
```

Comment:
ServerType is not really a proper name, a server is one of the following type
- Official 
- Private
That was what the MMO bool indicated, perhaps we should just have a bool to indicate if the server is official or not?

Now what we do have is a `ServerSourceProviderType` e.g. what provider made this server entry appear in the server list.
Currently we have 3 ways a server can appear in this list.
- it's an official server
- it's a community private server from the official community server list
- the server was added by a player to the list.

And in the future the `SS3` standard will be added in two different implementations so that adds 2 other types. and perhaps we could add https://screeps.online/ as a third source, depending on api access and such.